### PR TITLE
Fix troop counter position

### LINKED ISSUE #884: Creatures standing opposite each other shouldn't cover creature quantity
We can see quantity under Mummies here.
count

### PR DESCRIPTION
**Pull request by idshibanov**
Fixes #884 .

Corrected general position (both X and Y) applying offsets from original FRM BIN data files. Added additional check if there's unit in front and adjusting x offset accordingly.

image

### CODE REVIEW

File: src/fheroes2/battle/battle_interface.cpp [hunk 1]
```diff
@@ -1319,36 +1319,27 @@ void Battle::Interface::RedrawTroopSprite( const Unit & b )
...
 
-    s32 sx = 0;
-    s32 sy = 0;
+    const int tileInFront = isReflected ? unit.GetHeadIndex() - 1 : unit.GetHeadIndex() + 1;
 
-    const bool isReflected = b.isReflect();
-    const bool isWide = b.isWide();
-    sy = rt.y + rt.h - bar.h() - ( isReflected ? 21 : 9 );
+    s32 sx = rt.x + ( isReflected ? -7 : rt.w - 13 );
+    s32 sy = rt.y + rt.h - bar.h() - ( isReflected ? 23 : 11 );
```
[the remark this answers is not in the document]
Comment (idshibanov) [reply src/fheroes2/battle/battle_interface.cpp:1331]: It is exactly -11 and -23 in the original code that's why I adjusted. Looks identical to me when I did the eye test.

File: src/fheroes2/battle/battle_interface.cpp [hunk 2]
```diff
@@ -1319,36 +1319,27 @@ void Battle::Interface::RedrawTroopSprite( const Unit & b )
...
 {
-    const Rect & rt = b.GetRectPosition();
-    const Sprite & bar = AGG::GetICN( ICN::TEXTBAR, GetIndexIndicator( b ) );
+    const Rect & rt = unit.GetRectPosition();
+    const Sprite & bar = AGG::GetICN( ICN::TEXTBAR, GetIndexIndicator( unit ) );
+    const bool isReflected = unit.isReflect();
 
-    s32 sx = 0;
-    s32 sy = 0;
+    const int tileInFront = isReflected ? unit.GetHeadIndex() - 1 : unit.GetHeadIndex() + 1;
```
[the remark this answers is not in the document]
Comment (idshibanov) [reply src/fheroes2/battle/battle_interface.cpp:1328]: Hmm `isValidIndex` was supposed to deal with that but I forgot index is not coordinate. Good catch, will fix :)

[the remark this answers is not in the document]
Comment (idshibanov) [reply src/fheroes2/battle/battle_interface.cpp]: Should be good now, using Board::GetIndexDirection instead that will return -1 index in such case. Then the validation will fail.

**Comment by ihhub** [comment]
@idshibanov , thank you very much for fixing this issue!

**Pull request merged**